File: clean/node/src/infrastructure/router.ts
import express from "express";
import { TasksController } from "../interfaces/controllers/TasksController";
import { UsersController } from "../interfaces/controllers/UsersController";
import { MysqlConnection } from "./MysqlConnection";

const mysqlConnection = new MysqlConnection();
const tasksController = new TasksController(mysqlConnection);
const usersController = new UsersController(mysqlConnection);

let router = express.Router();

router.get("/tasks", async (req: express.Request, res: express.Response) => {
  let results = await tasksController.findAllTasks(req, res);
  res.send(results);
});

router.get(
  "/tasks/:id",
  async (req: express.Request, res: express.Response) => {
    let result = await tasksController.findTask(req, res);
    res.send(result);
  }
);

router.post("/tasks", async (req: express.Request, res: express.Response) => {
  let result = await tasksController.createTask(req, res);
  res.send(result);
});

router.patch(
  "/tasks/:id",
  async (req: express.Request, res: express.Response) => {
    let result = await tasksController.updateTask(req, res);
    res.send(result);
  }
);

router.delete(
  "/tasks/:id",
  async (req: express.Request, res: express.Response) => {
    let result = await tasksController.deleteTask(req, res);
    res.send(result);
  }
);

router.post("/users", async (req: express.Request, res: express.Response) => {
  let result = await usersController.createUser(req, res);
  res.send(result);
});

router.get("/users", async (req: express.Request, res: express.Response) => {
  let result = await usersController.findUser(req, res);
  res.send(result);
});

export default router;

File: clean/go/src/infrastructure/router.go
// Frameworks & Driversレイヤー

package infrastructure

import (
	gin "github.com/gin-gonic/gin"

	"ca/src/interfaces/controllers"
)

var Router *gin.Engine

func init() {
	router := gin.Default()

	userController := controllers.NewUserController(NewSqlHandler())
	taskController := controllers.NewTaskController(NewSqlHandler())

	router.POST("/users", func(c *gin.Context) { userController.CreateUser(c) })
	router.GET("/users", func(c *gin.Context) { userController.IndexUser(c) })
	router.GET("/users/:id", func(c *gin.Context) { userController.ShowUser(c) })
	router.GET("/tasks", func(c *gin.Context) { taskController.IndexTask(c) })
	router.POST("/tasks", func(c *gin.Context) { taskController.CreateTask(c) })
	router.GET("/tasks/:id", func(c *gin.Context) { taskController.ShowTask(c) })

	Router = router
}
